cmd/soroban-cli/src/rpc/mod.rs

```diff
@@ -626,22 +627,26 @@ soroban config identity fund {address} --helper-url <url>"#
     pub async fn prepare_transaction(
         &self,
         tx: &Transaction,
-    ) -> Result<(Transaction, Vec<DiagnosticEvent>), Error> {
+    ) -> Result<(Transaction, Option<RestorePreamble>, Vec<DiagnosticEvent>), Error> {
         tracing::trace!(?tx);
         let sim_response = self
             .simulate_transaction(&TransactionEnvelope::Tx(TransactionV1Envelope {
                 tx: tx.clone(),
                 signatures: VecM::default(),
             }))
             .await?;
-
         let events = sim_response
             .events
             .iter()
             .map(DiagnosticEvent::from_xdr_base64)
             .collect::<Result<Vec<_>, _>>()?;
-
-        Ok((assemble(tx, &sim_response)?, events))
+        Ok((
+            assemble(tx, &sim_response)?,
+            sim_response
+                .restore_preamble
+                .filter(|p| !p.transaction_data.is_empty()),
```

Comment:
maybe I'm misunderstanding the code here, but I don't think it should ever be empty if the `restore_preamble` is present in the rpc response

Reply:
Peeking into the output from other RPC tests it seems that an empty `TransactionData` is being sent when it should be ignored.

Reply:
Rebased to your commit and removed it! Thanks!